src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -203,25 +192,25 @@ public void Prepare(CancellationToken token)
                     return;
                 }
 
-                ConfigFileInformation info = keyValue.Value;
-                Debug.Assert(!string.IsNullOrWhiteSpace(info.NewFilePath), "Expected to be set during registration time");
-                
-                this.sourceControlledFileSystem.QueueFileWrite(info.NewFilePath, () =>
+                var info = keyValue.Value;
+                Debug.Assert(!string.IsNullOrWhiteSpace(info.FilePath), "Expected to be set during registration time");
+
+                sourceControlledFileSystem.QueueFileWrite(info.FilePath, () =>
                 {
-                    var ruleSetDirectoryPath = Path.GetDirectoryName(info.NewFilePath);
+                    var ruleSetDirectoryPath = Path.GetDirectoryName(info.FilePath);
```

Comment:
This is still here since SolutionBindingOperation has a Commit method so I couldn't easily to move it into IBindingConfigFile.Save

Reply:
It looks like we only need the `FilePath` here so the solution binder can create the folder if required.

The solution binder should already have ensured the solution-level binding folder exists. If any other sub-folders are required then creating them should be the responsibility of the language-specific binder (as will be the case when we add the C#/VB additional files).

Then, we could remove `FilePath` from the interface and just call `info.Save()` here.

I wouldn't do it now, but we should definitely look at whether we can remove `FilePath` from the interface later.